cli/src/commands/diff.rs

```diff
@@ -0,0 +1,526 @@
+use clap::Parser as ClapParser;
+use codec::Decode;
+use color_eyre::eyre::eyre;
+use frame_metadata::RuntimeMetadataPrefixed;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use crate::utils::FileOrUrl;
+use color_eyre::owo_colors::OwoColorize;
+use jsonrpsee::client_transport::ws::Uri;
+use scale_info::form::PortableForm;
+use scale_info::Variant;
+use subxt_codegen::utils::MetadataVersion;
+use subxt_metadata::{
+    ConstantMetadata, Metadata, PalletMetadata, RuntimeApiMetadata, StorageEntryMetadata,
+    StorageEntryType,
+};
+
+/// Explore the differences between two nodes
+///
+/// # Example
+/// ```
+/// subxt diff ./artifacts/polkadot_metadata_small.scale ./artifacts/polkadot_metadata_tiny.scale
+/// subxt diff ./artifacts/polkadot_metadata_small.scale wss://rpc.polkadot.io:443
+/// ```
+#[derive(Debug, ClapParser)]
+pub struct Opts {
+    node1: String,
+    node2: String,
+    #[clap(long, short)]
+    version: Option<MetadataVersion>,
+    #[clap(long, short)]
+    pallet: Option<String>,
```

Comment:
This isn't used AFAIU, what's intention behind it? Just compare a single pallet in the metadata?

If we want this, it would be better with Vec<String> to compare a one or more pallets